src/test/java/teammates/test/cases/datatransfer/StudentAttributesFactoryTest.java

```diff
@@ -15,110 +15,102 @@
+        EnrollException ee = assertThrows(EnrollException.class, () -> new StudentAttributesFactory(headerRow));
+        assertEquals(StudentAttributesFactory.ERROR_HEADER_ROW_FIELD_MISSED + ": <mark>Email</mark>",
+                ee.getMessage());
+    }
+
+    @Test
+    public void testConstructor_repeatedSingleColumn_fail() throws Exception {
 
         ______TS("Failure case: repeated required columns");
-        headerRow = "name \t email \t team \t comments \t name";
-        try {
-            new StudentAttributesFactory(headerRow);
-            signalFailureToDetectException();
-        } catch (EnrollException e) {
-            assertEquals(StudentAttributesFactory.ERROR_HEADER_ROW_FIELD_REPEATED, e.getMessage());
-        }
+        String headerRow = "name \t email \t team \t comments \t name";
+        EnrollException ee = assertThrows(EnrollException.class, () -> new StudentAttributesFactory(headerRow));
+        assertEquals(StudentAttributesFactory.ERROR_HEADER_ROW_FIELD_REPEATED, ee.getMessage());
+    }
+
+    @Test
+    public void testConstructor_repeatedMultipleColumns_fail() throws Exception {
 
         // adding this for complete coverage
         ______TS("Failure case: repeated required columns");
-        headerRow = "name \t email \t team \t comments \t section \t email \t team \t comments \t section ";
-        try {
-            new StudentAttributesFactory(headerRow);
-            signalFailureToDetectException();
-        } catch (EnrollException e) {
-            assertEquals(StudentAttributesFactory.ERROR_HEADER_ROW_FIELD_REPEATED, e.getMessage());
-        }
+        String headerRow = "name \t email \t team \t comments \t section \t email \t team \t comments \t section ";
+        EnrollException ee = assertThrows(EnrollException.class, () -> new StudentAttributesFactory(headerRow));
+        assertEquals(StudentAttributesFactory.ERROR_HEADER_ROW_FIELD_REPEATED, ee.getMessage());
 
         // remaining cases have been implicitly tested in testMakeStudent()
     }
 
     @Test
-    public void testMakeStudent() throws EnrollException {
-        StudentAttributesFactory saf = new StudentAttributesFactory();
-        String line = null;
-        String courseId = null;
-
-        StudentAttributes studentCreated;
-
+    public void testMakeStudent_emptyRow_fail() throws EnrollException {
         ______TS("Failure case: empty row");
-        line = "";
-        courseId = "SAFT.courseId";
-        try {
-            saf.makeStudent(line, courseId);
-            signalFailureToDetectException();
-        } catch (EnrollException e) {
-            assertEquals(StudentAttributesFactory.ERROR_ENROLL_LINE_EMPTY, e.getMessage());
-        }
+        StudentAttributesFactory saf = new StudentAttributesFactory();
+        String line = "";
+        String courseId = "SAFT.courseId";
+        EnrollException ee = assertThrows(EnrollException.class, () -> saf.makeStudent(line, courseId));
```

Comment:
Ensure line 91 is executed. In the previous versions of JUnit, anything below the exception-causing line is ignored silently.

Reply:
That's older JUnit which we're moving on from. `assertThrows` has the ability of capturing (only) the desired exception without breaking the program flow.